drafts/str4d-stratum/draft1.rst

```diff
@@ -0,0 +1,256 @@
+Abstract
+========
+
+This ZIP describes the Zcash variant of the Stratum protocol, used by miners to
+communicate with mining pool servers.
+
+
+Specification
+=============
+
+``mining.subscribe()``
+----------------------
+
+Request::
+
+    {"id": 1, "method": "mining.subscribe", "params": ["CONNECT_HOST", CONNECT_PORT, "MINER_USER_AGENT", "SESSION_ID"]}\n
+
+- ``CONNECT_HOST`` (str): The host string used by the miner.
+
+  - ``pool.example.com``
+
+- ``CONNECT_PORT`` (int): The port used by the miner.
```

Comment:
No, CONNECT_PORT is supposed to contain only the part of connection string given to miner by its user. If pool mining endpoint is stratum.mybestpool.com:3337, then CONNECT_HOST is "stratum.mybestpool.com" and CONNECT_PORT is 3337. It has nothing to do with GBT or any extra software running on miner. The motivation is the same as for CONNECT_HOST, but for consistency with mining.reconnect() is the URL split into parts.